Automattic/wp-calypso
# New Onboarding Launch: fix plans data-store integration

#### Changes proposed in this Pull Request

* Fix domain step to handle custom domain selection correctly. https://cloudup.com/cSsUZC30Rpg
  - re-use more of the useDomainSelection hooks
  - de-select plan and mark Plan step as incomplete when custom domain is picked and an existing Free plan was selected
* Fix plan step to disable _Free_ plan when a custom domain is selected. https://cloudup.com/c322tg7jwSg
* TypeScript fixes and `useSelect` refactors into same hook

#### Context
With the recent Plans data-store refactoring in #48790, Focused launch was integrated correctly (see changes for [Summary view](https://github.com/Automattic/wp-calypso/pull/48790/files#diff-bae2915b59f2e567e179b1ed48bdcb4d210fbc99442732fa6252d32ef7c2c181) and [Details view](https://github.com/Automattic/wp-calypso/pull/48790/files#diff-98b4f16e8ff2302875bc1e7e0c2c8cb8b678f90223dfbf71fd400cf5ddbdf854)) but post-Gutenboarding Launch needs some patches.

#### Testing instructions

* Create a site using `/new`
* Sandbox the site and open Launch flow from editor
* Confirm the subdomain and then select Free Plan
* Go back and select a custom domain => Plan step should be set as incomplete and plan selection lost
* Advance to the Plan step => Free plan should be disabled
* [ ] Smoke test the whole Step by Step flow, as this PR also refactors a lot of its code
* [ ] Check all Step by Step files to make sure that there's no more TypeScript errors
